scripts/web/utils/pdf_generator.py

```
@@ -0,0 +1,260 @@
+    pdf.set_text_color(148, 163, 184)
+    pdf.cell(0, 6, _safe_ascii(
+        f"Generated by GradeAI Pro  |  {datetime.now().strftime('%Y-%m-%d %H:%M:%S')}"
+    ), align="C")
+
+    # Save & update FIFO index
+    safe_name    = re.sub(r"[^\w\-]", "_", student)[:40]
+    pdf_filename = f"{run_id}_{safe_name}.pdf"
+    pdf_path     = REPORTS_DIR / pdf_filename
+    try:
+        pdf.output(str(pdf_path))
+    except Exception:
```

Ruff (0.15.11)

[warning] Do not catch blind exception: `Exception`

(BLE001)